src/app-layer-parser.c

```diff
@@ -140,6 +140,8 @@ struct AppLayerParserState_ {
      * we don't need a var per direction since we don't log a transaction
      * unless we have the entire transaction. */
     uint64_t log_id;
+    /* Count of number of transactions */
+    uint64_t tx_cnt;
```

Comment:
In fact, what the code is adding to the tv counter at each pass in the parser the difference of tx on a flow between now and the previous pass. So we take into account the new TX and we have to keep the tx_cnt between the pass.